src/ssl.c

```diff
@@ -16522,21 +16522,33 @@ static const char* wolfssl_ffdhe_name(word16 group)
  */
 const char* wolfSSL_get_curve_name(WOLFSSL* ssl)
 {
+    const char* cName = NULL;
+
     if (ssl == NULL)
         return NULL;
+
 #ifdef HAVE_FFDHE
-    if (ssl->namedGroup != 0)
-        return wolfssl_ffdhe_name(ssl->namedGroup);
+    if (ssl->namedGroup != 0) {
+        cName = wolfssl_ffdhe_name(ssl->namedGroup);
+    }
+#endif
+
+    if (cName == NULL) {
+#ifdef HAVE_CURVE25519
+        if (ssl->ecdhCurveOID == ECC_X25519_OID) {
+            cName = "X25519";
+        }
 #endif
+
 #ifdef HAVE_ECC
-    if (ssl->ecdhCurveOID == 0)
-        return NULL;
-    if (ssl->ecdhCurveOID == ECC_X25519_OID)
-        return "X25519";
-    return wc_ecc_get_name(wc_ecc_get_oid(ssl->ecdhCurveOID, NULL, NULL));
-#else
-    return NULL;
+        if (ssl->ecdhCurveOID != 0 && cName == NULL) {
```

Comment:
Let's not nest the logic here. Instead do a `cName == NULL` check at each point. Currently with this code if `HAVE_CURVE25519` is not defined then cName NULL check is done twice.

Reply:
Great note! Thanks.